File: backend/src/main/java/com/tenpista/challenge/backend/domain/model/Transaction.java
package com.tenpista.challenge.backend.domain.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.OffsetDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Transaction {
    private Long id;
    private Integer amount;
    private String commerceName;
    private String tenpistaName;
    private OffsetDateTime transactionDate;
    private OffsetDateTime createdAt;
}

File: mfe-transactions/src/types/transaction.ts
export interface Transaction {
  id: number;
  amount: number;
  commerce_name: string;
  tenpista_name: string;
  transaction_date: string;
  created_at: string;
}

export interface TransactionPage {
  content: Transaction[];
  total_elements: number;
  total_pages: number;
  page: number;
  size: number;
}

export interface TransactionCreateRequest {
  amount: number;
  commerce_name: string;
  tenpista_name: string;
  transaction_date: string;
}
